spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,51 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+      table: DeltaTableV2,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String],
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTable = withActiveSession(sparkSession) {
+    val (targetRelation, targetPath) = if (new Path(target).isAbsolute()) {
+      (UnresolvedRelation(TableIdentifier(target, Some("delta"))), Some(target))
+    } else {
+      val tableIdentifier = sparkSession
+        .sessionState
+        .sqlParser
+        .parseTableIdentifier(target)
+      (UnresolvedRelation(tableIdentifier), None)
+    }
+    val sourceRelation = DataSourceV2Relation.create(table, None,
+      table.getTableIdentifierIfExists.map(id => Identifier.of(id.database.toArray, id.table)))
+
+    val maybeTimeTravelSource = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
+      TimeTravel(
+        sourceRelation,
+        timestampAsOf.map(Literal(_)),
+        versionAsOf,
+        None
+      )
+    } else {
+      sourceRelation
+    }
+
+    val clone = CloneTableStatement(maybeTimeTravelSource, targetRelation, false, replace, true,
+      properties.toMap, targetPath)
+
+    val qe = sparkSession.sessionState.executePlan(clone)
+
+    // call `QueryExecution.toRDD` to trigger the execution of commands.
+    SQLExecution.withNewExecutionId(qe, Some("clone delta table"))(qe.toRdd)
+
+    if (targetPath.isDefined) {
+      DeltaTable.forPath(sparkSession, targetPath.get)
```

Comment:
```suggestion
      DeltaTable.forPath(sparkSession, target)
```

This would do the trick, same for the `DeltaTable.forName(sparkSession, target)` also, so much simplified

Reply:
Much